### ISSUE TITLE
Fetch module dest= relative path when using roles

### ISSUE THREAD

**Issue by dbohdan** [ansible/ansible-modules-core#2157]
##### Issue Type:
- Bug Report
##### Ansible Version:

```
ansible 1.9.3
  configured module search path = None
```
##### Ansible Configuration:

N/A
##### Environment:

N/A
##### Summary:

Unlike `copy` or `synchronize`, `fetch` does not make the current role's `files` directory the reference point for relative local paths.

(This was previously discussed at https://github.com/ansible/ansible/issues/4566.)
##### Steps To Reproduce:

Run a task like the following from `roles/myrole/tasks/main.yml`:

``` Yaml
- name: back up remote file
  fetch:
    src: /remote/foo
    dest: foo
    flat: yes
```
##### Expected Results:

`/remote/foo` being copied to `roles/myrole/files/`.
##### Actual Results:

`/remote/foo` is copied to the directory of the playbook that includes the role (e.g., `playbook.yml`).

**Comment by robynbergeron** [comment]
[module: files/fetch.py]

**Comment by robynbergeron** [comment]
Hi @dbohdan -- thanks for submitting this bug report, much appreciated!

@ansible/core team -- pinging you to let you know about this bug report.

**Comment by ansibot** [comment]
@ansible, ping. This issue is still waiting on your response.
[click here for bot help](https://github.com/jctanner/ansibullbot/blob/ISSUE_TRIAGE_CLASS/ISSUE_HELP.md)

**Comment by tima** [comment]
@dbohdan: This seems like correct behavior to me. Roles should be under some sort of version control or being managed thru ansible-galaxy. It's not a good idea to be storing arbitrary files being fetched in there that it makes sense that the fetch module does not use the role path with relative paths. 

I checked synchronize with the latest stable release (2.1.1.0) and it does not support this (incorrect) relative path logic you experienced in pre-2.0 releases before all the refactoring and rewriting. 

wontfix

**Comment by dbohdan** [comment]
@tima: I see. Personally, I don't mind this being resolved one way or another as long as `copy`, `synchronize` and `fetch` all behave consistently in the end. Good to know that 2.0 brings that.

**Comment by tima** [comment]
@dbohdan from my quick tests copy, fetch and synchronize are working as designed. A relative copy or sync src is relative to the roles files/ path. A relative fetch or sync dest is relative to the roles files/ path. All other relative paths are relative to the playbook running the role.

**Comment by dbohdan** [comment]
@tima I understood your previous comment to mean that `fetch` in 2.x still used `dest`s relative to the playbook directory, not the role's `files/` dir, and that `sync` has been updated to match it. If, instead, it is the case that

> A relative fetch or sync dest is relative to the roles files/ path.

this matches what I describe as expected in the bug report and what two other users expected `fetch` to do in https://github.com/ansible/ansible/issues/4566. If so, this bug is not wontfix but fixed.

**Comment by tima** [comment]
Right. Sorry that came out wrong. Let's try again.

A relative copy or sync src is relative to the roles files/ path. All other relative paths are relative to the playbook running the role.

**Comment by dbohdan** [comment]
@tima Ah, okay. That clears it up.
